packages/toast/src/ToastContainer/ToastContainer.tsx

```diff
@@ -0,0 +1,342 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Transition, TransitionGroup } from 'react-transition-group';
+
+import { cx } from '@leafygreen-ui/emotion';
+import {
+  useBackdropClick,
+  useDynamicRefs,
+  useIdAllocator,
+  useMutationObserver,
+  useStateRef,
+} from '@leafygreen-ui/hooks';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import {
+  createSyntheticEvent,
+  createUniqueClassName,
+  keyMap,
+} from '@leafygreen-ui/lib';
+import Portal from '@leafygreen-ui/portal';
+import { transitionDuration } from '@leafygreen-ui/tokens';
+
+import { TOAST_CONSTANTS } from '../constants';
+import { InternalToast } from '../InternalToast';
+import { ToastId, ToastStack, useToast } from '../ToastContext';
+
+import {
+  NotificationBar,
+  notificationBarTransitionStyles,
+} from './NotificationBar';
+import {
+  containerCollapsingStyles,
+  containerExpandedStyles,
+  getContainerInteractedStyles,
+  getContainerStatefulStyles,
+  getToastHoverStyles,
+  getToastTransitionStyles,
+  getToastUnhoveredStyles,
+  scrollContainerExpandedStyles,
+  scrollContainerStyles,
+  scrollContainerTransitionOutStyles,
+  toastContainerStyles,
+  toastContainerVisibleStyles,
+} from './ToastContainer.styles';
+import {
+  getDividedStack,
+  useToastHeights,
+  useToastTimers,
+  useToastTransitions,
+} from './utils';
+
+export const toastPortalClassName = createUniqueClassName('toast-portal');
+
+/**
+ * ToastContainer is responsible for rendering the stack of toasts provided
+ */
+export const ToastContainer = ({ stack }: { stack: ToastStack }) => {
+  const { popToast, getToast } = useToast();
+  const regionId = useIdAllocator({ id: 'lg-toast-region' });
+  const toastContainerRef = useRef<HTMLDivElement>(null);
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+  const getToastRef = useDynamicRefs<HTMLDivElement>({ prefix: 'toast' });
+  const { theme } = useDarkMode();
+  // Track hover state
+  const [isHovered, setHoveredState] = useState(false);
+  const setHovered = () => setHoveredState(true);
+  const setUnhovered = () => setHoveredState(false);
+  // Track intended expand state
+  const [shouldExpand, setShouldExpand, getShouldExpand] = useStateRef(false);
+  const expandToasts = () => setShouldExpand(true);
+  const collapseToasts = () => setShouldExpand(false);
+  // Track stack size
+  const stackSize = stack.size;
```

Comment:
nit - doesn't really feel necessary to have another const here

Reply:
yeah that's fair 👍